### ISSUE TITLE
Flatten a metrics object into loggable scalars, as log_params does for hyperparameters

### ISSUE THREAD

**Issue by alirezanobakht13** [alirezanobakht13/runsnap#2]
`runsnap.log_params()` flattens a Pydantic model into one param per scalar leaf.
There is no equivalent for the *metrics* side: a training iteration produces a
small object of scalar arrays (losses, gradient norm, KL, counters) that has to
be turned into plain Python numbers before it can be logged or charted.

The helper that had to remain in `pomdp`:

```python
def scalars(module: eqx.Module) -> dict[str, Any]:
    """Flatten a module of scalar arrays into loggable Python numbers."""
    result = {}
    for field in dataclasses.fields(module):
        value = getattr(module, field.name)
        if isinstance(value, eqx.Module):
            result.update(scalars(value))          # nested -> flat keys
        else:
            value = value.item()
            result[field.name] = (
                None if isinstance(value, float) and not math.isfinite(value) else value
            )
    return result
```

Two details are the actual work, and both are the kind of thing a tracking
library should settle once: nested records flatten to dotted/flat keys the same
way `log_params` already flattens nested models, and non-finite floats become
`None` rather than a `NaN` that MLflow stores and charts as a real value.

**Suggestion:** a `runsnap.flatten_metrics(obj)` (or extending `flatten_model`'s
traversal to dataclasses/equinox modules and array leaves) so a per-iteration
diagnostics object can go straight to `log_metrics` / TensorBoard.

Raised from `pomdp`, runsnap's first consumer.

**Comment by alirezanobakht13** [comment]
Implemented in a4f85c2 as `runsnap.flatten_metrics(obj)`: mappings, dataclasses (equinox modules included), and Pydantic models flatten to dotted keys; one-element arrays become Python numbers, booleans 0/1, and `None`/strings are dropped. One deliberate difference from this request: non-finite values are kept as `NaN` rather than `None`, so a divergence shows up in the curve. pomdp will replace its `scalars()` helper with it.
